preupg/xml_to_json.py

```diff
@@ -0,0 +1,416 @@
+        """
+        attribute_data = []
+        for ele in element:
+            data = self.getDataFromElement(ele, attribute)
+            attribute_data.append(data)
+        return {self.removeNamespace(element[0]): {attribute: attribute_data}}
+
+    def getDataFromElement(self, element, attribute):
+        if attribute is "text":
+            return element.text
+        else:
+            return element.get(attribute)
+
+    def wrapElementAttrInDict(self, element, attr):
+        """
+        element, attr = "foo", "text"
+        >>> "{foo: "text from foo element"}"
+
+        element, attr = "foo", "boo"
+        >>> "{foo: "value of attribute boo"}"
+        """
+        if attr == "text":
+            if element.text is None:
+                raise AttrNotFound("Cannot find text in {0} element"
+                                   .format(self.removeNamespace(element)))
+            return {self.removeNamespace(element[0]): element.text}
+        else:
+            if element.get(attr) is None:
+                raise AttrNotFound("Cannot find {0} attribute in {1} element"
+                                   .format(attr, self.removeNamespace(element)))
+            return {attr: element.get(attr)}
+
+    def removeNamespace(self, element):
+        return element.tag.replace(self.namespace[self.namespace_key], "")
+
+    def parseCheckImportOutput(self, text):
+        """
+        Divide text from <check-import> element
+        according to RISK and LOG statements
+        """
+        risk = []
+        log = []
+        paragraphs = text.split('\n')
+        for paragraph in paragraphs:
+            if "preupg.log" in paragraph:
+                log.append(paragraph)
+            elif "preupg.risk" in paragraph:
+                risk.append(paragraph)
+            else:
+                pass
+        parsed_dict = {'risk': risk, 'log': log}
+        if not risk:
+            parsed_dict.pop('risk')
+        if not log:
+            parsed_dict.pop('log')
+        if not parsed_dict:
+            raise EmptyText
+        return parsed_dict
+
+    def parseSetValue(self, element, data):
+        idref = self.getDataFromElement(element, "idref")
+        requested_keys = map(lambda key: "xccdf_preupg_value_"+str(key),
+                             GLOBAL_DIC_VALUES.keys())
+        if idref in requested_keys:
+            return {idref: data}
+        else:
+            return None
+
+    def addOptToResultDict(self, dictionary, where_in_dict, option_dict):
+        if(where_in_dict == "general" or where_in_dict == "results"):
+            dictionary[where_in_dict].update(option_dict)
+
+    def createJson(self, dictionary):
+        return json.dumps(dictionary, indent=4, separators=(',', ': '))
+
+    def getJsonText(self):
+        return self.createJson(self.result_dict)
+
+
+class MissingElement(Exception):
+    """
+    Element in xml file does not exist
+    """
+    pass
+
+
+class EmptyText(Exception):
```

Comment:
Name the exceptions so they end with either _Exception_ or _Error_, e.g. EmptyTextException.